moshi/src/test/java/com/squareup/moshi/PromoteNameToValueTest.java

```diff
@@ -252,7 +252,7 @@
       writer.value(true);
       fail();
     } catch (IllegalStateException e) {
-      assertThat(e).hasMessage("Nesting problem.");
+      assertThat(e).hasMessage("Boolean cannot be used as a map key in JSON.");
```

Comment:
This catch block uses `e` instead of `expected`, but I'd actually prefer the latter, `expected` vs `ignored` seems like a good naming convention. I think it's a nit though